### PR TITLE
Fix make generate

### PR DESCRIPTION
**Pull request by drpaneas**
* Introduces `go.tools` which can be used as a placeholder to put `go get`-able stuff that are required for the project, such as the `openapi-gen`. As such the `go.mod` gets tricked and uses those as **indirect** dependencies.
* Introduces `make download` that runs `go mod download` to fetch the golang universe.
* Introduces `make install-tools` which is running `go install $tool` against all the lines at `tools.go` and has `make download` as a prereq.
* Removes `lint` and `isclean` as they are undefined in the makefile - obsoleted.

PS: I am not including `golangci-lint` as this is handled by the `boilerplate` code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -341,3 +341,13 @@ clean-operator: ## Clean Operator
...
+
+.PHONY: download
+download: ## Download dependencies
+	@echo Downloading go.mod dependencies ...
+	@go mod download
+
+.PHONY: install-tools
+install-tools: download ## Get tools
+	@echo Installing tools from tools.go ...
+	@cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go install %
```
Comment (dastergon) [inline Makefile:353]: Can you add a line break here?

Comment (drpaneas) [reply Makefile]: thanks, I already did it ;)
FYI: next time feel free to use https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request so I can click-and-commit on your proposals ;)

**Comment by 2uasimojo** [comment]
> * Introduces `go.tools` which can be used as a placeholder to put `go get`-able stuff that are required for the project, such as the `openapi-gen`. As such the `go.mod` gets tricked and uses those as **indirect** dependencies.

You shouldn't need to do this for generate/build/test/lint related deps The [boilerplate backing image](https://github.com/openshift/boilerplate/blob/eb577a390b8cd99e45be60d635d7ca9cdcab62f3/config/build.sh) already contains [openapi-gen](https://github.com/openshift/boilerplate/blob/eb577a390b8cd99e45be60d635d7ca9cdcab62f3/config/build.sh#L66-L70) and other necessaries. Trying to install and run that stuff locally *might* work, but eventually will cause you headaches due to platform and version discrepancies. Instead run those things through [container-make](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/boilerplate/_lib/container-make), which uses said backing image, which contains said tools.

> * Removes `lint` and `isclean` as they are undefined in the makefile - obsoleted.

[lint](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/boilerplate/openshift/golang-osd-operator/standard.mk#L189-L191) and [isclean](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/boilerplate/openshift/golang-osd-operator/standard.mk#L84-L86) are defined in the boilerplate standard.mk.

**Comment by drpaneas** [comment]
I wasn't aware of this container. So, it seems we have everything. Please adjust the documentation to include this information.

**Comment by 2uasimojo** [comment]
> Please adjust the documentation to include this information.

The main README [references](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/README.md#boilerplate) the boilerplate [convention's README](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/boilerplate/openshift/golang-osd-operator/README.md), which [mentions container-make](https://github.com/openshift/aws-account-operator/blob/bef554871bbf67bc8d633d43805260d3dc1ade2a/boilerplate/openshift/golang-osd-operator/README.md#local-testing). If you have a suggestion for making it more obvious, I'm all :ear:s

**Pull request closed without merging**